mreg_cli/outputmanager.py

```diff
@@ -114,28 +115,42 @@ class OutputManager:
     def __new__(cls):
         if cls._instance is None:
             cls._instance = super().__new__(cls)
-
-            # Recording related attributes. These must come first as they may
-            # be encountered when calling clear(). Note that these attributes are
-            # not reset between commands, which is why they are not themselves
-            # manipulated in clear().
-            cls._recording: bool = False
-            cls._filename: str = None
-            cls._recorded_data: List[str] = []
-
             cls._instance.clear()
+            cls._instance.recording_clear()
 
         return cls._instance
 
     def clear(self) -> None:
         """Clears the object."""
-        self._lines = []
+        self._output: [str] = []
         self._filter_re = None
-        self._negate = False
-        self._command = None
-        self.command = None
+        self._filter_negate = False
+        self._command_executed = None
+        self._command_issued = None
+
+        self._ok: [str] = []  # This is typically commands that went OK but returned no content
+        self._warnings: [str] = []
+        self._errors: [str] = []
+
+        self._api_requests: [str] = []
+
+        self._time_started: datetime.datetime = datetime.datetime.now()
+
+    def recording_clear(self) -> None:
+        """Clears the recording data."""
+        self._recorded_data = []
```

Comment:
```suggestion
        self._recorded_data: List[Dict[str, Any]] = []
```



One could also define a custom type for the dict produced by `recording_entry()`, and use that as a type instead:


 ```py
class RecordingEntry(TypedDict): ...

def recording_entry(self) -> RecordingEntry: ...
     
# and then:
self._recorded_data: List[RecordingEntry] = []

Reply:
Yes! Good idea! Apart from supporting older pythons... I have... hacked my way around it. Ish.